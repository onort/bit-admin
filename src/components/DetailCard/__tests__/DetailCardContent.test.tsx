import React from "react"
import { shallow } from "enzyme"

import { DetailCardContent } from "../"

const testContent = "Test content."
const testClassName = "testClassName"

describe("<DetailCardContent />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <DetailCardContent>
        <span>{testContent}</span>
      </DetailCardContent>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should display passed content", () => {
    const wrapper = shallow(
      <DetailCardContent>
        <span>{testContent}</span>
      </DetailCardContent>
    )
    expect(wrapper.find("span").text()).toBe(testContent)
  })

  it("should be able to have custom className", () => {
    const wrapper = shallow(
      <DetailCardContent className={testClassName}>
        <span>{testContent}</span>
      </DetailCardContent>
    )
    expect(wrapper.find(`.${testClassName}`).exists()).toBe(true)
  })
})
