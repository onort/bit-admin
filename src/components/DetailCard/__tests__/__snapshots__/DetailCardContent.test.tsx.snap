// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailCardContent /> should match snapshot 1`] = `
<div
  className="contentContainer"
>
  <span>
    Test content.
  </span>
</div>
`;

exports[`<DetailCardContent /> should match snapshot 2`] = `"<div class=\\"contentContainer\\"><span>Test content.</span></div>"`;
